apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexMatcherServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+        this.epochTime =  Objects.requireNonNull( epochTime,"epochTime is null");        
+    }
+    
+    /**
+     * Start matcher service
+     */
+    
+    @Override
+    public void initialize() {
+        log.debug("DexMatcherService : initialization routine");
+        this.lastMatchedOffer = null;
+    }
+
+    /**
+     * Stop matcher service
+     */
+    
+    @Override
+    public void deinitialize() {
+        log.debug("DexMatcherService : deinitialization routine");        
+    }
+    
+    
+    /**
+     * currency-specific validation (Ethereum)
+     * @param DexOffer  offer to validate
+     */ 
+    private boolean validateOfferETH(DexOffer offer) {
+        // TODO: add validation routine
+        return true;         
+    }
+
+    /**
+     * currency-specific validation (Apollo)
+     * @param DexOffer  offer to validate
+     */ 
+    private boolean validateOfferAPL(DexOffer offer) {
+        // TODO: add validation routine
+        return true;         
+    }
+    
+    /**
+     * currency-specific validation (Pax)
+     * @param DexOffer  offer to validate
+     */ 
+    private boolean validateOfferPAX(DexOffer offer) {
+        // TODO: add validation routine
+        return true;         
+    }
+    
+    
+    /**
+     * Common validation routine for offer  
+     * @param DexOffer  offer - offer to validate
+     */ 
+    private boolean validateOffer( DexOffer offer) {
+        
+        DexCurrencies curr = offer.getOfferCurrency(); 
+        log.debug("currency: {}", curr );
+        
+        switch (curr) {
+            case APL: return validateOfferAPL(offer);
+            case ETH: return validateOfferETH(offer);
+            case PAX: return validateOfferPAX(offer);
+            default: return false;
+        }        
+        
+    }
+    
+            
+
+    /**
+     * Core event for matcher - when offer is created, it is called back     
+     * @param DexOffer  myOffer - some data of offer, that is being created
+     * @param DexOffer  hisOffer - the most suitable offer the Deal
+     */ 
+    
+    private void onOfferMatch ( DexOffer myOffer, DexOffer hisOffer) {
+        log.debug("DexMatcherService.onOfferMatch callback ");
+        log.debug("match..  id: {}, offerCurrency: {}, offerAmount: {}, pairCurrency: {}, pairRate: {} ", hisOffer.getAccountId(), hisOffer.getOfferCurrency(), hisOffer.getOfferAmount(), hisOffer.getPairCurrency(), hisOffer.getPairRate() ); 
+    }
+    
+        
+    /**
+     * Core event for matcher - when offer is created, it is called back     
+     * @param offerType  Type of the offer. (BUY/SELL) 0/1
+     */
+
+    public void onCreateOffer( DexOffer createdOffer ) {
+        
+        log.debug("DexMatcherServiceImpl:onCreateOffer()");
+        
+        OfferType type = null;
+    
+        // vice versa - if we create buy offer we need to look up sell             
+        if ( createdOffer.getType().equals(OfferType.SELL)) type = OfferType.BUY;
```

Comment:
You can use this construction.
OfferType counterOfferType = createdOffer.getType().isSell() ? OfferType.BUY : OfferType.SELL;

Reply:
Done as described